### PR TITLE
Update version management section to reflect semver based versions

### PR DESCRIPTION
**Pull request by dietmarw**
versionBuild is no longer advised to use and hence removed.

### CODE REVIEW

File: Modelica/package.mo [hunk 1]
```diff
@@ -2335,7 +2336,7 @@ more of the following changes.
 <li> Introducing a new name in the public section of a class
          (model, package, ...) or in any section of a partial class is <strong>not</strong> allowed.
          Since otherwise, a user might use this new name and when storing its model
-         and loading it with an older build-version, an error would occur.</li>
+         and loading it with an older bugfix-version, an error would occur.</li>
```
Comment (beutlich) [inline Modelica/package.mo]:
```suggestion
         and loading it with an older bug-fix version, an error would occur.</li>
```

**Comment by dietmarw** [comment]
@henrikt-ma You might want to add your (valuable) input above to the MCPs in question since they might get lost here in this PR.

**Comment by dietmarw** [comment]
I'll update the documentation with your input but would still mention semver since that is were we got the original idea from anyway. So might as well state that it is inspired by that.

**Comment by beutlich** [comment]
> 
> 
> I'll update the documentation with your input but would still mention semver since that is were we got the original idea from anyway. So might as well state that it is inspired by that.

Please update, too:

https://github.com/modelica/ModelicaStandardLibrary/blob/3dfe8836cbacdd4a1fd9ea4b383259bfd6657c3e/Modelica/package.mo#L2359

**Comment by henrikt-ma** [comment]
> Instead state we use semantic versioning (MAJOR.MINOR.BUGFIX), where bug-fixes are restricted as described in this document; and state that it will be clarified later.

Sounds like the right thing to do given the tight schedule.  I hope that we will later be able to interpret this as MSL 4 conforming to the future definitions of MCP0016, even for the versions of MSL 4 that were released before MCP0016 was accepted.  (If we would have had more time, I would have suggested to actually state in MSL 4.0.0 what we think should go into MCP0016 — that way, there would be no need in the future to retroactively change the version semantics of already released MSL versions.)

**Comment by dietmarw** [comment]
I've updated the description and have hopefully covered the concerns.

**Pull request merged**